staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -978,6 +998,10 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
 		if err != nil {
 			return err
 		}
+		if !hasDNSMetricsContext {
+			req = req.WithContext(httptrace.WithClientTrace(req.Context(), newDNSMetricsTrace(ctx)))
```

Comment:
if we add teh context before the for loop we don't need the bool, do we?

Reply:
scratch teh comment, I didn't see the request is created here

Reply:
Does the context setup need to happen inside `newHTTPRequest` to cover requests made via Watch/Stream? It looks like you'd also be able to drop the bool and conditional in that case.

Reply:
@benluddy good catch, addressed!